Update toplayer visiblity on workspace switch

Consolatis opened the pull request:
Fixes:
- #1158

Only had a very short test nested.

I am also not sure if `view.c` is actually the correct place for the new `view_update_top_layer()` function:
- As it depends on a set of views it could also belong to `desktop.c`.
- As it toggles the visibility of a specific layershell layer it could also belong to `layers.c`.

Consolatis commented on src/view.c:
```diff
@@ -1097,6 +1102,33 @@ view_toggle_fullscreen(struct view *view)
+view_update_top_layer(struct server *server)
+{
+	struct view *view;
+	struct output *output;
+
+	uint32_t top = ZWLR_LAYER_SHELL_V1_LAYER_TOP;
```
remove empty line

jlindgren90 commented:
Didn't test but LGTM. It looks like this would also correctly handle the case where you have two fullscreen windows and close one of them. :+1:

johanmalm commented:
LGTM. Have tested and seems to work great.

Nice to increase usage of `for_each_view()`

Agree, `desktop.c` feel like a better home. Even if we're dragging some layer-stuff it's all a function of the views.

Suggest `s/view_update_top_layer/desktop_update_top_layer_visibility/`

Consolatis commented:
> Agree, `desktop.c` feel like a better home. Even if we're dragging some layer-stuff it's all a function of the views.

Moved to `desktop.c`

> Suggest `s/view_update_top_layer/desktop_update_top_layer_visibility/`

Changed

johanmalm commented:
Really nice patch. Thanks!
